app/upgrades/v8/upgrades.go

```diff
@@ -49,6 +52,65 @@ func FixBankMetadata(ctx sdk.Context, keepers *keepers.AppKeepers) error {
 	return nil
 }
 
+func QuicksilverFix(ctx sdk.Context, keepers *keepers.AppKeepers) error {
+	// Refund stuck coins from ica address
+	sourceAddress, err := sdk.AccAddressFromBech32("cosmos13dqvh4qtg4gzczuktgnw8gc2ewnwmhdwnctekxctyr4azz4dcyysecgq7e")
+
+	if err != nil {
+		return errors.New("invalid source address")
+	}
+	destinationAddress, err := sdk.AccAddressFromBech32("cosmos1jc24kwznud9m3mwqmcz3xw33ndjuufnghstaag")
+	if err != nil {
+		return errors.New("invalid destination address")
+	}
+
+	// Get balance from stuck address and subtract 1 uatom sent by bad actor
+	sourceBalance := keepers.BankKeeper.GetBalance(ctx, sourceAddress, "uatom")
+	refundBalance := sourceBalance.SubAmount(sdk.NewInt(1))
+	keepers.BankKeeper.SendCoins(ctx, sourceAddress, destinationAddress, sdk.NewCoins(refundBalance))
+
+	// Get connection to quicksilver chain
+	connectionId, err := getConnectionIdForChainId(ctx, keepers, "quicksilver-1")
+	if err != nil {
+		return err
+	}
+
+	// Close channels
+	closeChannel(keepers, ctx, connectionId, "icacontroller-cosmoshub-4.deposit")
+	closeChannel(keepers, ctx, connectionId, "icacontroller-cosmoshub-4.withdrawal")
+	closeChannel(keepers, ctx, connectionId, "icacontroller-cosmoshub-4.performance")
+	closeChannel(keepers, ctx, connectionId, "icacontroller-cosmoshub-4.delegate")
```

Comment:
Maybe similar comment as with the connection ID. If we know for sure that the channels that we want to close have channel IDs `channel-462`, `channel-463`, `channel-464` and `channel-465` then we could use those IDs directly instead. I see that [all those channels are in connection `connection-707`](https://rest.cosmos.directory/cosmoshub/ibc/core/channel/v1/connections/connection-707/channels), are `OPEN` and have the counterparty port IDs mentioned here (there's also a `channel-466` with counterparty port ID `icacontroller-cosmoshub-4.delegate` in state `TRYOPEN`, but I think that would not be picked up by the call to `GetActiveChannelID`).

Reply:
We'll change the code to use the specific channel IDs, instead of retrieving the channel with GetActiveChannelID